### PR TITLE
Performance improvements

### PR DESCRIPTION
**Pull request by Tom18314**
Just some performance improvements. The database thing really helps alot. We used this changes for over an year so I am sure they work.

### CODE REVIEW

File: EpicSpawners-Plugin/src/main/java/com/craftaro/epicspawners/EpicSpawners.java [hunk 1]
```diff
@@ -104,8 +104,15 @@ public void onPluginDisable() {
         if (!this.spawnerManager.wasConfigModified()) {
             this.saveToFile();
         }
-        this.particleTask.cancel();
-        this.spawnerCustomSpawnTask.cancel();
+
+        if (Bukkit.getScheduler().isCurrentlyRunning(this.particleTask.getTaskId())) {
```
Comment (SpraxDev) [inline EpicSpawners-Plugin/src/main/java/com/craftaro/epicspawners/EpicSpawners.java:108]: I'm not sure if `#isCurrentlyRunning` is the correct method here...
If I understand the [JavaDocs](https://hub.spigotmc.org/javadocs/spigot/org/bukkit/scheduler/BukkitScheduler.html#isCurrentlyRunning(int)) for it correctly, for repeating-tasks it actually tells you if they are running right now and not when they are idle and waiting for their next scheduled execution

Have you experienced weird behavior that could be related to that? Is there an issue in canceling a task when it has already been cancelled in the past?
> A repeating task might not be running currently, but will be running in the future.

File: EpicSpawners-Plugin/src/main/java/com/craftaro/epicspawners/EpicSpawners.java [hunk 2]
```diff
@@ -104,8 +104,15 @@ public void onPluginDisable() {
...
+            this.particleTask.cancel();
+        }
+
+        if (Bukkit.getScheduler().isCurrentlyRunning(this.spawnerCustomSpawnTask.getTaskId())) {
```
Comment (SpraxDev) [inline EpicSpawners-Plugin/src/main/java/com/craftaro/epicspawners/EpicSpawners.java:112]: Same as some lines further up:

> I'm not sure if `#isCurrentlyRunning` is the correct method here...
> If I understand the [JavaDocs](https://hub.spigotmc.org/javadocs/spigot/org/bukkit/scheduler/BukkitScheduler.html#isCurrentlyRunning(int)) for it correctly, for repeating-tasks it actually tells you if they are running right now and not when they are idle and waiting for their next scheduled execution
> 
> Have you experienced weird behavior that could be related to that? Is there an issue in canceling a task when it has already been cancelled in the past?
> > A repeating task might not be running currently, but will be running in the future.

**Review by SpraxDev — CHANGES_REQUESTED** [review]
One thing I am not sure about - Rest looks great to me :)
Thanks for your interest in contributing

**Comment by SpraxDev** [comment]
Hi @Tom18314, any updates on this?

**Comment by Tom18314** [comment]
@SpraxDev no, never had any issues with this.

**Comment by SpraxDev** [comment]
@ceze88 I think you wanted to merge this together with some other change?

**Comment by ceze88** [comment]
I already made those changes

**Pull request merged**